lib/queue.rb

```
@@ -1,16 +1,39 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = @back = 0
+    end
+    
+    if @front == @back
+    end
+
+    @store[@back] = element 
+    @back = (@back + 1) % @store.length 
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @back
+      return "queue is empty"
```

It would be better to raise an error in this circumstance.